worker/src/enclave/worker_api_direct_server.rs

```diff
@@ -227,93 +227,61 @@ pub fn handle_direct_invocation_request(req: DirectWsServerRequest) -> Result<()
-                    continue_watching = false;
-                }
-                _ => {}
-            };
-            // update response
-            result.status = DirectRequestStatus::TrustedOperationStatus(status_update);
-            event.result = result.encode();
-            client_event
-                .client
-                .send(serde_json::to_string(&event).unwrap())
-                .unwrap();
-
-            if !continue_watching {
-                client_event.client.close(CloseCode::Normal).unwrap();
-            }
-        } else {
-            continue_watching = false;
-        }
-        if !continue_watching {
-            watch_list.remove(&hash);
-        }
-    }
-
     sgx_status_t::SGX_SUCCESS
 }
 
 #[no_mangle]
-pub unsafe extern "C" fn ocall_send_status(
-    hash_encoded: *const u8,
-    hash_size: u32,
-    status_encoded: *const u8,
-    status_size: u32,
+pub unsafe extern "C" fn ocall_send_response_with_uuid(
+    request_id_encoded: *const u8,
+    request_id_size: u32,
+    uuid_encoded: *const u8,
+    uuid_size: u32,
 ) -> sgx_status_t {
-    let status_slice = slice::from_raw_parts(status_encoded, status_size as usize);
-    let mut hash_slice = slice::from_raw_parts(hash_encoded, hash_size as usize);
-    if let Ok(hash) = Hash::decode(&mut hash_slice) {
-        // Aquire watched list lock
-        let mutex = load_watched_list().unwrap();
-        let mut guard = mutex.lock().unwrap();
-        if let Some(client_response) = guard.get_mut(&hash) {
-            let mut response = &mut client_response.response;
-
-            // create return value
-            // TODO: Signature?
-            let submitted =
-                DirectRequestStatus::TrustedOperationStatus(TrustedOperationStatus::Submitted);
-            let result = RpcReturnValue::new(status_slice.to_vec(), false, submitted);
+    let mut request_id_slice = slice::from_raw_parts(request_id_encoded, request_id_size as usize);
+    let uuid_slice = slice::from_raw_parts(uuid_encoded, uuid_size as usize);
+    if let Ok(request_id) = RequestId::decode(&mut request_id_slice) {
+        let mutex = if let Some(mutex) = load_watched_list() {
+            mutex
+        } else {
+            return sgx_status_t::SGX_ERROR_UNEXPECTED;
+        };
+        let mut guard = if let Ok(value) = mutex.lock() {
+            value
+        } else {
+            return sgx_status_t::SGX_ERROR_UNEXPECTED;
+        };
+        let submitted = DirectRequestStatus::Ok;
+        let result = RpcReturnValue::new(uuid_slice.to_vec(), false, submitted);
 
-            // update response
+        if let Some(client_response) = guard.get_mut(&request_id) {
+            let mut response = &mut client_response.response;
             response.result = result.encode();
             client_response
                 .client
-                .send(serde_json::to_string(&response).unwrap())
+                .send(serde_json::to_string(uuid_slice).unwrap())
```

Comment:
Above you're updating the response, but here you're still sending the uuid

Reply:
sorry .. chianged it